### PR TITLE
Add Button to expand recents.  Simplify localStorage.

### LINKED ISSUE #591: [BUG] Recents get truncated at 20
Need a better way of managing the recents list:

* `updatedAt` metadata value to actually sort by recent
* Ability to "show more" or something at the bottom of the list kinda like girder classic.

I don't want to use a data table, they're too clunky and I like the clean feel of this page.

### PR DESCRIPTION
**Pull request by subdavis**
At one point earlier today I managed to write 400 lines of code including a full schema migration to support a new field called `lastUpdated`.  I discarded all those changes and re-wrote it to be as simple as possible.

* Still sorts by creation time, which may not be ideal.
* Simple more/less toggle.  If we get more than like 100, we'll need more options, but this is fine for now.
* Added some code to remove most of the of unnecessary cost in storage and serialization from large image sequences.
* Backed out that dumb thing where I show either 1 or a stack of images.  Now all image sequences are just a stack.\
*

fixes #591

### CODE REVIEW

**Comment by subdavis** [comment]
Forgot to request.

File: client/platform/desktop/frontend/store/dataset.ts [hunk 1]
```diff
@@ -42,7 +45,17 @@ function load(): JsonMeta[] {
...
+    ...meta,
+    /**
+     * Erase image lists from meta object stored in recents.
+     * Saves space and serialization cost since these parts of
+     * the recents object aren't used in this way.
+     */
+    imageData: [],
+    originalImageFiles: [],
+    transcodedImageFiles: [],
+  });
```
Comment (BryonLewis) [inline client/platform/desktop/frontend/store/dataset.ts:58]: I know it's a pain but I'm wondering if we should really be using a different type here given this comment and the one above about the `localCachedMeta` being different from `JsonMeta`?

**Review by BryonLewis — APPROVED** [review]
Works perfectly fine.  I have that one comment but I don't feel strongly about it.  The end result is clear because of the comments and I imagine any future issue would begin with following the data path and noting that the Image lists get removed.

File: client/platform/desktop/frontend/store/dataset.ts (hunk 1 above)
Comment (subdavis) [reply client/platform/desktop/frontend/store/dataset.ts:58]: Valid.  I originally did this.  There's a couple of ways to hack "subtraction" to extends an interface by removing fields in typescript.  I ended up not liking the results and just went with the comment.

This actually goes against my philosophy about never telling lies to the type system, but it just felt so weird to redeclare a core interface here.

What approach would you prefer?

Comment (BryonLewis) [reply client/platform/desktop/frontend/store/dataset.ts:58]: I looked into the subtraction stuff quickly when I was doing my review and I had the same reaction.

I say a new type that has the overlapping data expected for the frontend cached dataset to work.  It clearly separates them as `localCacheMeta` is derived from `JsonMeta` and not the same thing.  

It also gives a place to store future optional data that is relevant to the front-end display but not required to be stored in `JsonMeta`.  Like we may want to edit the `load` of recents process to indicate to the user that the source file/folder is missing (I don't know if we want to test for sourcedata for each item in recents on each app load, just an example).  That could be added as an optional `localCacheMeta` property easily showing the user that some of their recents have been moved or deleted outside of DIVE.

**Review by BryonLewis — APPROVED** [review]
looks good

**Pull request merged**